sim/rogue/runes.go

```diff
@@ -361,17 +361,18 @@ func (rogue *Rogue) applyRollingWithThePunches() {
 		return
 	}
 
-	statDeps := make([]*stats.StatDependency, 6) // 5 stacks + zero condition
-	for i := 1; i < 6; i++ {
-		statDeps[i] = rogue.NewDynamicMultiplyStat(stats.Health, 1.0+.06*float64(i))
-
+	rogue.rollingWithThePunchesBonusHealthStackMultiplier += 0.06
+	rogue.rollingWithThePunchesMaxStacks += 5
+	statDeps := make([]*stats.StatDependency, rogue.rollingWithThePunchesMaxStacks+1) // 5 stacks + zero condition
+	for i := 1; i < rogue.rollingWithThePunchesMaxStacks+1; i++ {
```

Comment:
I think this does not work with the property incremented in phase 8 6p aura because it's created at `applyRWTPRune` and never recalculated. Was it done to minimize compute efforts in OnStackChanges @kayla-glick ? I think to make it truly dynamic with the 6p aura, I need to move the 
`rogue.NewDynamicMultiplyStat(stats.Health, 1.0+rogue.rollingWithThePunchesBonusHealthStackMultiplier*float64(i))` 
inside the aura like this

`aura.Unit.DisableDynamicStatDep(sim, rogue.NewDynamicMultiplyStat(stats.Health, 1.0+rogue.rollingWithThePunchesBonusHealthStackMultiplier*float64(oldStacks)))`

Reply:
I don't know if NewDynamicMultiplyStat is costly to call on every stack change

Reply:
@fynnen oh yeah you're right. You would have to increment the attributes outside of the aura in the set code, which is totally fine to do in this case. Usually as long as you're not trying to reference a spell or aura in the to-level code because chances are they don't exist yet.

> I don't know if NewDynamicMultiplyStat is costly to call on every stack change
You can't create stat dependencies once the sim has gotten to that point. If you tried it would do a `panic`

Reply:
I might not be understanding the lifecycle correctly, but is `applyRollingWithThePunches` called after all auras are set? Because if it's done _before_, `statDeps` will never be recalculated right?

Reply:
@fynnen this line setting up the stat dependencies is called before auras are initialized. That's why we'd have to move the set bonus increment the variable to outside of the aura

Reply:
It worked by moving the set bonus increments outside of the aura! Thanks!